src/OpalCompiler-Tests/OCBytecodeToASTCacheTest.class.st

```diff
@@ -60,21 +61,17 @@ OCBytecodeToASTCacheTest >> testLowerThanFirstBCOffsetAccessTest [
 { #category : #tests }
 OCBytecodeToASTCacheTest >> testNodeForBCOffsetRangeTest [
 	"As we associate each node to each possible bytecode offset that can refer to it,
-	we have to check that associations are consistent between a range and a node"
+	we have to check that associations are all pointing to a node"
 
 	| pcRange |
 	pcRange := 0 to: cache lastBcOffset.
-	pcRange do: [ :pc | 
-		self
-			assert: (cache nodeForPC: pc)
-			identicalTo: (compiledMethod sourceNodeForPC: pc) ]
+	pcRange do: [ :pc | self assert: ((cache nodeForPC: pc) isKindOf: RBProgramNode)]
```

Comment:
I agree in the end that is better, however we are only testing that it maps to a node but not the correct one.
I would do a brute check: for each pc, check that it is the correct node.
Maybe we can generate it?

Reply:
I add this to the backlog, I would like to merge the current state